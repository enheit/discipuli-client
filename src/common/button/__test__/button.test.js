import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import Button from '../button';

describe('Button component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Button type="button" positive>Button</Button>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should be rendered with class \'.button--positive\'', function() {
    expect(shallow(<Button type="button" positive>Button</Button>)
      .is('.button--positive'))
      .toBe(true);
  });

  it('should be rendered with class \'.button--negative\'', function() {
    expect(shallow(<Button type="button" negative>Button</Button>)
      .is('.button--negative'))
      .toBe(true);
  });

  it('should be rendered with disabled state and class \'.button--disabled\'', function() {
    expect(shallow(<Button type="button" disabled>Button</Button>)
      .is('.button--disabled'))
      .toBe(true);
  });


  it('should be clickable', () => {
    const mockCallBack = jest.fn();

    const button = shallow((<Button onClick={mockCallBack}>Button</Button>));
    button.find('button').simulate('click');
    expect(mockCallBack.mock.calls.length).toEqual(1);
  });
});
