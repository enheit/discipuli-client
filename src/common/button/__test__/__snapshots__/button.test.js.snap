// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders correctly 1`] = `
<button
  className="button button--positive"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  Button
</button>
`;
